packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -66,29 +66,35 @@ export class TriggerManager extends Component {
       account
     })
 
-    let folder
-    if (konnectors.needsFolder(konnector)) {
-      const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
-        accounts.getLabel(account)
-      )}`
-
-      folder =
-        (await statDirectoryByPath(path)) || (await createDirectoryByPath(path))
-
-      await addPermission(konnector, konnectors.buildFolderPermission(folder))
-      await addReferencesTo(konnector, [folder])
+    try {
+      let folder
+
+      if (konnectors.needsFolder(konnector)) {
+        const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
+          accounts.getLabel(account)
+        )}`
+
+        folder =
+          (await statDirectoryByPath(path)) ||
+          (await createDirectoryByPath(path))
+
+        await addPermission(konnector, konnectors.buildFolderPermission(folder))
+        await addReferencesTo(konnector, [folder])
+      }
+
+      const trigger = await createTrigger(
+        triggers.buildAttributes({
+          account,
+          cron: cron.fromKonnector(konnector),
+          folder,
+          konnector
+        })
+      )
+
+      return await this.launch(trigger)
+    } catch (error) {
+      return this.handleError(error)
     }
```

Comment:
I think this business logic should be outside the React component.

Reply:
We agree, the next step is to start using Models.